test/server/listener_manager_impl_test.cc

```diff
@@ -1267,20 +1267,19 @@ TEST_F(ListenerManagerImplWithRealFiltersTest, OriginalDstTestFilter) {
 
   class OriginalDstTestConfigFactory : public Configuration::NamedListenerFilterConfigFactory {
   public:
-    OriginalDstTestConfigFactory() : option_(std::make_shared<Network::MockSocketOption>()) {}
-
     // NamedListenerFilterConfigFactory
     Configuration::ListenerFilterFactoryCb
     createFilterFactoryFromProto(const Protobuf::Message&,
                                  Configuration::ListenerFactoryContext& context) override {
-      EXPECT_CALL(*option_, setOption(_, Network::Socket::SocketState::PreBind))
+      auto option = std::make_unique<Network::MockSocketOption>();
+      EXPECT_CALL(*(option.get()), setOption(_, Network::Socket::SocketState::PreBind))
```

Comment:
`EXPECT_CALL(*option, setOption` should work, no?

Reply:
You are right, an explicit `get()` is not needed here. Will simplify this in a commit ASAP.